puppet/modules/slave/manifests/pbuilder_setup.pp

```diff
@@ -7,31 +7,41 @@
   $backports  = false,
   $nodesource = true,
   $puppetlabs = true,
+  $pbuilder_type = 'pbuilder',
 ) {
 
-  pbuilder { $name:
-    ensure    => $ensure,
-    arch      => $arch,
-    release   => $release,
-    methodurl => $apturl,
-  }
-
-  file { "/etc/pbuilder/${name}/apt.config/sources.list.d":
-    ensure  => bool2str($ensure == present, 'directory', 'absent'),
+  case $pbuilder_type {
+    'pbuilder': {
+      pbuilder { $name:
+        ensure    => $ensure,
+        arch      => $arch,
+        release   => $release,
+        methodurl => $apturl,
+      }
+      $update_pbuilder = "update_pbuilder_${name}"
+    }
+    'cowbuilder': {
+      pbuilder::cowbuilder { "${name}":
+        ensure     => $ensure,
+        arch       => $arch,
+        dist       => $release,
```

Comment:
this will become `release` in https://github.com/camptocamp/puppet-pbuilder/pull/13